cmake/openthread_extensions.cmake

```diff
@@ -0,0 +1,121 @@
+    endif()
+  endforeach()
+
+  set(OPENTHREAD_CONFIG_FILE
+    "${CMAKE_BINARY_DIR}/openthread_lib_configuration.txt"
+    CACHE STRING "File containg OpenThread build configuration parameters.")
+  FILE(WRITE ${OPENTHREAD_CONFIG_FILE} ${OPENTHREAD_SETTINGS})
+
+  set(OPENTHREAD_LIB_FTD "${OPENTHREAD_SRC_DIR}/core/libopenthread-ftd.a"
+    CACHE STRING "FTD")
+  message(STATUS "OPENTHREAD_LIB_FTD=${OPENTHREAD_LIB_FTD}")
+
+  set(OPENTHREAD_LIB_MTD "${OPENTHREAD_SRC_DIR}/core/libopenthread-mtd.a"
+    CACHE STRING "MTD")
+
+  set(OPENTHREAD_LIB_RADIO "${OPENTHREAD_SRC_DIR}/core/libopenthread-radio.a"
+    CACHE STRING "RADIO")
+
+  set(OPENTHREAD_LIB_CLI_FTD "${OPENTHREAD_SRC_DIR}/cli/libopenthread-cli-ftd.a"
+    CACHE STRING "CLI-FTD")
+
+  set(OPENTHREAD_LIB_CLI_MTD "${OPENTHREAD_SRC_DIR}/cli/libopenthread-cli-mtd.a"
+    CACHE STRING "CLI-MTD")
+
+  set(OPENTHREAD_LIB_NCP_FTD "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-ncp-ftd.a"
+    CACHE STRING "NCP-FTD")
+
+  set(OPENTHREAD_LIB_NCP_MTD "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-ncp-mtd.a"
+    CACHE STRING "NCP-MTD")
+
+  set(OPENTHREAD_LIB_RCP "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-rcp.a"
+    CACHE STRING "RCP")
+
+  nrfxlib_calculate_lib_path(lib_path)
+
+  set(OPENTHREAD_DST_DIR
+    "${NRFXLIB_DIR}/openthread/${lib_path}/v${CONFIG_OPENTHREAD_THREAD_VERSION}"
+    CACHE STRING "The nrfxlib directory with OpenThread Libraries to be overwritten.")
+endfunction(openthread_libs_configuration_write)
+
+if(CONFIG_NET_L2_OPENTHREAD AND CONFIG_OPENTHREAD_SOURCES)
+  message(DEBUG "Building OT from sources, config file will be generated.")
+  openthread_libs_configuration_write()
+
+  set(OPENTHREAD_HEADERS_DIR "${ZEPHYR_OPENTHREAD_MODULE_DIR}/../include")
+
+  add_custom_target(install_openthread_libraries
+    COMMAND ${CMAKE_COMMAND} -E make_directory "${OPENTHREAD_DST_DIR}"
+    COMMAND ${CMAKE_COMMAND} -E copy "${OPENTHREAD_LIB_FTD}" "${OPENTHREAD_DST_DIR}"
+    COMMAND ${CMAKE_COMMAND} -E copy "${OPENTHREAD_LIB_MTD}" "${OPENTHREAD_DST_DIR}"
+    COMMAND ${CMAKE_COMMAND} -E copy "${OPENTHREAD_LIB_RADIO}" "${OPENTHREAD_DST_DIR}"
+    COMMAND ${CMAKE_COMMAND} -E copy "${OPENTHREAD_LIB_CLI_FTD}" "${OPENTHREAD_DST_DIR}"
+    COMMAND ${CMAKE_COMMAND} -E copy "${OPENTHREAD_LIB_CLI_MTD}" "${OPENTHREAD_DST_DIR}"
+    COMMAND ${CMAKE_COMMAND} -E copy "${OPENTHREAD_LIB_NCP_FTD}" "${OPENTHREAD_DST_DIR}"
+    COMMAND ${CMAKE_COMMAND} -E copy "${OPENTHREAD_LIB_NCP_MTD}" "${OPENTHREAD_DST_DIR}"
+    COMMAND ${CMAKE_COMMAND} -E copy "${OPENTHREAD_LIB_RCP}" "${OPENTHREAD_DST_DIR}"
+    COMMAND ${CMAKE_COMMAND} -E copy_directory
+      ${OPENTHREAD_HEADERS_DIR}
+      ${NRFXLIB_DIR}/openthread/include
+    COMMAND ${CMAKE_COMMAND} -E copy
+     ${OPENTHREAD_CONFIG_FILE}
+     ${NRFXLIB_DIR}/openthread/
+    DEPENDS ${OPENTHREAD_LIB_FTD}
+    )
+
+    set_property(TARGET zephyr_property_target
+    APPEND PROPERTY install_openthread_libraries_DEPENDENCIES
+    install_openthread_libraries
+    )
+endif()
```

Comment:
extension cmake files should only contain extension functions, not code to be executed during CMake configure phase itself.